src/Nethermind/Nethermind.Consensus/Processing/OneTimeProcessor.cs

```diff
@@ -45,7 +46,7 @@ public Task StopAsync(bool processRemainingBlocks = false)
                 }
                 finally
                 {
-                    _readOnlyDbProvider.ClearTempChanges();
+                    _worldState.Reset();
```

Comment:
I still don't understand why stick to to passing the whole `IWorldState` here to call only the reset after processing. I'd add something like `IResettable` and implement it in `IWorldState` or a component that provides `IDiposable` that is created only for the block processing with `using var scope = scope.Create()` and hide the reset behind.

Reply:
Because IWorldState already have a `Reset`. Adding an `IResettable` does not add any value at the moment, and it will make it harder to know what is being reset. It can't add any scope at the moment.

Reply:
It looks we have different povs on knowing what to reset. My take is that processor should not know what is reset here. It should just know that it's a scoping that is required. We can easily add scope locally. Let me provide some option below:

```csharp
namespace Nethermind.Consensus.Processing
{
    public interface IScopeProvider
    {
        IDisposable Begin();
    }
    
    public class OneTimeChainProcessor : IBlockchainProcessor
    {
        public ITracerBag Tracers => _processor.Tracers;

        private readonly IBlockchainProcessor _processor;
        private readonly IScopeProvider_worldState;

        private object _lock = new();

        public OneTimeChainProcessor(IScopeProvider worldState, IBlockchainProcessor processor)
        {
            _worldState = worldState ?? throw new ArgumentNullException(nameof(worldState));
            _processor = processor ?? throw new ArgumentNullException(nameof(processor));
        }

        public void Start()
        {
            _processor.Start();
        }

        public Task StopAsync(bool processRemainingBlocks = false)
        {
            return _processor.StopAsync(processRemainingBlocks);
        }

        public Block? Process(Block block, ProcessingOptions options, IBlockTracer tracer)
        {
            lock (_lock)
            {
                using var scope = _worldState.Begin();
                return _processor.Process(block, options, tracer);
            }
        }

        public bool IsProcessingBlocks(ulong? maxProcessingInterval)
        {
            return _processor.IsProcessingBlocks(maxProcessingInterval);
        }

#pragma warning disable 67
        public event EventHandler<BlockProcessedEventArgs> BlockProcessed;
        public event EventHandler<BlockProcessedEventArgs> BlockInvalid;
        public event EventHandler<IBlockchainProcessor.InvalidBlockEventArgs>? InvalidBlock;
#pragma warning restore 67

        public void Dispose()
        {
            _processor?.Dispose();
        }
    }
}
```

Reply:
The `_worldState.Begin()` is being done in `ReadOnlyTxProcessingEnv` right now, outside of this class. I can't do it like that. I'm trying not to change things too much at the moment.